enhancements/automated-resource-request-scaling.md

```diff
@@ -0,0 +1,517 @@
+
+* This enhancement describes an approach for configuring OpenShift
+  clusters to run with lower CPU resource requests than its default
+  configuration.
+* Clusters built in this way should pass the same Kubernetes and
+  OpenShift conformance and functional end-to-end tests as single-node
+  deployments that have not been scaled down.
+* We have a goal of 4 hyperthreads today, but we do not know what
+  constraints we might be given in the future so we need a solution
+  that is not tied to the current known limit.
+
+### Non-Goals
+
+* This enhancement does not address memory resource requests.
+* This enhancement does not address ways to disable operators or
+  operands entirely.
+* Although the results of this enhancement may be useful for
+  single-node developer deployments, multi-node production
+  deployments, and Internet-of-things devices, those use cases are not
+  addressed directly in this document.
+
+## Proposal
+
+There are two aspects to manage scaling the control plane resource
+requests. Most of the cluster operators are deployed by the
+cluster-version-operator using static manifests that include the
+resource requests. Those operators then deploy other controllers or
+workloads dynamically. We need to account for both sets of components.
+
+
+
+This is where we get down to the nitty gritty of what the proposal actually is.
+
+### User Stories
+
+Detail the things that people will be able to do if this is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+
+Include a story on how this proposal will be operationalized:  lifecycled, monitored and remediated at scale.
+
+#### Story 1
+
+#### Story 2
+
+### Implementation Details/Notes/Constraints [optional]
+
+What are the caveats to the implementation? What are some important details that
+didn't come across above. Go in to as much detail as necessary here. This might
+be a good place to talk about core concepts and how they relate.
+
+### Risks and Mitigations
+
+What are the risks of this proposal and how do we mitigate. Think broadly. For
+example, consider both security and how this will impact the larger OKD
+ecosystem.
+
+How will security be reviewed and by whom? How will UX be reviewed and by whom?
+
+Consider including folks that also work outside your immediate sub-project.
+
+## Design Details
+
+### Open Questions
+
+1. The cluster-version-operator manages cluster operators using static
+   manifests. How would we make changes to the settings of Deployments
+   defined in those manifests?
+2. How do we express to the cluster operators that they should scale
+   their operands?
+3. How dynamic do we actually want the setting to be? Should the user
+   be able to change it at will, or is it a deployment-time setting?
+4. What happens during an upgrade? The new manifests from the new
+   release payload would have unscaled limits.
```

Comment:
Updates aren't all that special from the perspective of individual resources.  It's just a CVO reconciling manifests like the CVO always does, just from a different bag of manifests.  I expect 2nd-level operators are similar in how they roll out updates to their own operands.  However we have these limits applied in the steady state should apply on updates, although it will be easier to be sure once we have a steady-state plan.  And a plan for single-node updates ;).

Reply:
Yeah, I was thinking that if some other actor changes the requests, then the CVO gets a new manifest it would overwrite the value when applying the new thing, then the other actor would come a long and change it again. Maybe that's not what would happen, though, if the request fields are ignored when the manifest and resource are compared?

Reply:
yeah, if we teach the CVO to not care about these requests (which it would have to do in order to allow the non-CVO controller to manage them), the incoming CVO would continue to ignore the requests, so I don't expect any CVO stomping to take place, even on updates.

Reply:
OK, excellent, I'll update the text to take that into account.